addons/website_sale_comparison/static/src/js/website_sale_comparison.js

```diff
@@ -113,14 +115,17 @@ var ProductComparison = Widget.extend({
         }
     },
     add_new_products:function(product_id){
+        this.guard.exec(this._add_new_products.bind(this, product_id));
+    },
+    _add_new_products: function (product_id) {
         var self = this;
         $('.o_product_feature_panel').show();
         if (!_.contains(self.comparelist_product_ids, product_id)) {
             self.comparelist_product_ids.push(product_id);
             if(_.has(self.product_data, product_id)){
                 self.update_content([product_id], false);
             } else {
-                self.load_products([product_id]).then(function(){
```

Comment:
I am not sure to understand the original code but this commit adds a `return` here so the `update_cookie` which is after it won't be called anymore... Does this break anything, how did it work before ?

Reply:
the update_cookie update the browser cookies to add the newly compared product, it is necessary to do this after product are loaded, I just fixed my pr